staging/src/k8s.io/kubectl/pkg/cmd/config/get_contexts.go

```diff
@@ -142,19 +143,29 @@ func (o GetContextsOptions) RunGetContexts() error {
 			}
 		}
 	}
-	if o.showHeaders {
-		err = printContextHeaders(out, o.nameOnly)
-		if err != nil {
-			allErrs = append(allErrs, err)
-		}
-	}
 
 	sort.Strings(toPrint)
-	for _, name := range toPrint {
-		err = printContext(name, config.Contexts[name], out, o.nameOnly, config.CurrentContext == name)
-		if err != nil {
+
+	switch o.outputFormat {
```

Comment:
We don't need manual printing mechanism. Please see https://github.com/kubernetes/kubectl/blob/6096dfa3cf2f889a504332dc71380eedab8488f6/pkg/cmd/set/set_serviceaccount.go#L145-L146 as a example, `printer` contains `PrintObj` function that prints in a given format (that also supports json, yaml, etc.) for us freely.

Reply:
This cannot be used easily because `Context` is not a `runtime.Object`. Also there is no `ContextList` type. I think the main conceptual issue is that the code you point to is related to printing standard resources, which is not the case here it seems. The request is more like to get the command output in JSON, rather than any particular resource in JSON.